### PR TITLE
Add multi guests support in ansible_test

### PR DESCRIPTION
**Pull request by qcheng-redhat**
Add multi guests support in ansible_test
ID: 1891434

Signed-off-by: Qinghua Cheng <qcheng@redhat.com>

### CODE REVIEW

**Comment by qcheng-redhat** [comment]
@PaulYuuu @CongLi 

Please help review this patch.

File: qemu/tests/ansible_test.py [hunk 1]
```diff
@@ -15,18 +16,31 @@ def run(test, params, env):
...
+    4) Genreate inventory file
+    5) Generate the ansible-playbook command
+    6) Execute the playbook and verify the return status
 
     :param test: QEMU test object
     :param params: Dictionary with the test parameters
     :param env: Dictionary with test environment.
     """
+
+    def generate_invent_file(env):
```
Comment (PaulYuuu) [inline qemu/tests/ansible_test.py:28]: We don't need to generate the inventory file, use `",".join(ip_lst)` is enough.

File: qemu/tests/ansible_test.py [hunk 2]
```diff
@@ -41,15 +55,17 @@ def run(test, params, env):
     # Use this directory to copy some logs back from the guest
     test_harness_log_dir = test.logdir
 
-    vm = env.get_vm(params["main_vm"])
-    vm.verify_alive()
-    vm.wait_for_login()
-    guest_ip = vm.get_address()
+    for vm in env.get_all_vms():
+        vm.verify_alive()
+        vm.wait_for_login()
```
Comment (PaulYuuu) [inline qemu/tests/ansible_test.py:60]:
```suggestion
    guest_ip_list = []
    for vm in env.get_all_vms():
        vm.verify_alive()
        vm.wait_for_login()
        guest_ip_list.append(vm.get_address())
```

File: qemu/tests/ansible_test.py [hunk 3]
```diff
@@ -59,7 +75,7 @@ def run(test, params, env):
 
     ansible_cmd_options = ["ansible-playbook",
                            "-u {}".format(guest_user),
-                           "-i {},".format(guest_ip),
+                           "-i %s" % invent_file,
```
Comment (PaulYuuu) [inline qemu/tests/ansible_test.py:78]:
```suggestion
                           "-i {},".format(",".join(guest_ip_list)),
```

File: qemu/tests/ansible_test.py (hunk 1 above)
Comment (qcheng-redhat) [reply qemu/tests/ansible_test.py]: Ok. Updated.

File: qemu/tests/ansible_test.py (hunk 3 above)
Comment (qcheng-redhat) [reply qemu/tests/ansible_test.py]: Updated. Thanks

File: qemu/tests/ansible_test.py [hunk 4]
```diff
@@ -41,10 +41,11 @@ def run(test, params, env):
...
 
-    vm = env.get_vm(params["main_vm"])
-    vm.verify_alive()
-    vm.wait_for_login()
-    guest_ip = vm.get_address()
+    guest_ip_list = []
+    for vm in env.get_all_vms():
+        vm.verify_alive()
+        vm.wait_for_login()
+        guest_ip_list.append(vm.wait_for_get_address(0, 240))
```
Comment (PaulYuuu) [inline qemu/tests/ansible_test.py:48]: The guest has already logged in, so `vm.get_address()` would be more suitable.

Comment (qcheng-redhat) [reply qemu/tests/ansible_test.py]: Updated.

**Comment by PaulYuuu** [comment]
@qcheng-redhat Could you provide a test result?

**Comment by qcheng-redhat** [comment]
python3 ConfigTest.py --category=nested_x86_q35 --hostname=RHEL-8.3.0 --machine=q35 --customsparams="l2_guest_name=RHEL.8.3.0" --clone=no

(1/1) nested_multi_guests.Host_RHEL.m8.u3.v0.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.ansible_test.q35: PASS (201.53 s)

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**